src/jit/instr.cpp

```diff
@@ -1663,6 +1663,8 @@ void                CodeGen::inst_TT_RV(instruction   ins,
 #ifdef _TARGET_XARCH_
         assert(!instIsFP(ins));
 #endif
+
+#ifdef LEGACY_BACKEND
```

Comment:
Here too #if CPU_LONG_USES_REGPAIR?

Note that only x86/arm32 legacy backends defined CPU_LONG_USES_REGPAIR as 1.  So using 
#if CPU_LONG_USES_REGPAIR would be sufficient.